shared_model/backend/protobuf/query_responses/impl/proto_block_query_response.cpp

```diff
@@ -10,38 +10,28 @@
 namespace shared_model {
   namespace proto {
 
-    template <typename QueryResponseType>
-    BlockQueryResponse::BlockQueryResponse(QueryResponseType &&queryResponse)
-        : CopyableProto(std::forward<QueryResponseType>(queryResponse)),
+    BlockQueryResponse::BlockQueryResponse(TransportType &&block_query_response)
+        : proto_(std::move(block_query_response)),
           variant_{[this] {
-            auto &&ar = *proto_;
+            auto &ar = proto_;
             int which = ar.GetDescriptor()
                             ->FindFieldByNumber(ar.response_case())
                             ->index();
             return shared_model::detail::variant_impl<
                 ProtoQueryResponseVariantType::types>::
-                template load<ProtoQueryResponseVariantType>(
-                    std::forward<decltype(ar)>(ar), which);
+                template load<ProtoQueryResponseVariantType>(ar, which);
```

Comment:
could we change `ProtoQueryResponseVariantType` for `expected::Result<BlockResponse, BlockErrorResponse>` to avoid somewhat scary `shared_model::detail::variant_impl::...`?

Reply:
`Result` is a variant of two wrapped types. It will require the same deserialization call.

Reply:
I mean that the semantics of `expected::Result` seem to cover the needs of `ProtoQueryResponseVariantType`, which is either `BlockResponse` or `BlockErrorResponse`. This would make the approach more uniform and enable the use of the set of functions we have to make the handling of `Result`s more convenient.

Reply:
Considering that we might replace `expected::Result` with something else, this change might not be related to this pull request.